x-pack/plugins/security_solution/scripts/run_playwright/playwright.ts

```diff
@@ -0,0 +1,381 @@
+                }
+
+                await providers.loadAll();
+                const functionalTestRunner = new FunctionalTestRunner(
+                  log,
+                  config,
+                  EsVersion.getDefault()
+                );
+                const ftrEnv = await pRetry(() => functionalTestRunner.run(abortCtrl.signal), {
+                  retries: 1,
+                });
+                log.debug(
+                  `Env. variables returned by [functionalTestRunner.run()]:\n`,
+                  JSON.stringify(ftrEnv, null, 2)
+                );
+
+                // Normalized the set of available env vars in Playwright
+                const playwrightCustomEnv = {
+                  ...ftrEnv,
+                  // NOTE:
+                  // ELASTICSEARCH_URL needs to be created here with auth because SIEM Playwright setup depends on it. At some
+                  // points we should probably try to refactor that code to use `ELASTICSEARCH_URL_WITH_AUTH` instead
+                  ELASTICSEARCH_URL:
+                    ftrEnv.ELASTICSEARCH_URL ?? createUrlFromFtrConfig('elasticsearch', true),
+                  ELASTICSEARCH_URL_WITH_AUTH: createUrlFromFtrConfig('elasticsearch', true),
+                  ELASTICSEARCH_USERNAME:
+                    ftrEnv.ELASTICSEARCH_USERNAME ?? config.get('servers.elasticsearch.username'),
+                  ELASTICSEARCH_PASSWORD:
+                    ftrEnv.ELASTICSEARCH_PASSWORD ?? config.get('servers.elasticsearch.password'),
+                  FLEET_SERVER_URL: createUrlFromFtrConfig('fleetserver'),
+                  KIBANA_URL: baseUrl,
+                  KIBANA_URL_WITH_AUTH: createUrlFromFtrConfig('kibana', true),
+                  KIBANA_USERNAME: config.get('servers.kibana.username'),
+                  KIBANA_PASSWORD: config.get('servers.kibana.password'),
+                  IS_SERVERLESS: config.get('serverless'),
+                };
+
+                const envFilePath = path.resolve(
+                  __dirname,
+                  '..',
+                  '..',
+                  '..',
+                  '..',
+                  'test',
+                  'security_solution_playwright',
+                  '.env'
+                );
+
+                const envContent = Object.entries(playwrightCustomEnv)
+                  .map(([key, value]) => `${key}=${value}`)
+                  .join('\n');
+
+                fs.writeFileSync(envFilePath, envContent);
+
+                log.info(`
+                ----------------------------------------------
+                Playwright run ENV for file: ${filePath}
+                ----------------------------------------------
+                `);
+
+                const project = playwrightCustomEnv.IS_SERVERLESS ? 'serverless' : 'ess';
+
+                if (isOpen) {
+                  await execa.command(
+                    `npx playwright test --config ${playwrightConfigFilePath} --ui --project ${project}`,
```

Comment:
mind if we reference `./node_modules/bin/playwright` directly.  `npx` will install binaries if they don't exist, and it may not be the version we expect.